zipline/gens/brokers/broker.py

```diff
@@ -42,15 +42,15 @@ def time_skew(self):
         pass
 
     @abstractmethod
-    def order(self, asset, amount, limit_price, stop_price, style):
+    def order(self, asset, amount, style):
```

Comment:
This looks like a change of upstream API. Why do we need it? Can we avoid doing this?

Reply:
To be in sync with blotter. Changing the APIs are the only way to cleanly interface with blotter, here is why:
Initially the orders were intercepted in the LiveTradingAlgorithm class and never sent to blotter. I created the broker to easily work with that calling signature.

Now, moving to blotter the calls are not intercepted, but channeled through blotter.
Blotter uses style as opposed to limit_price and stop_price.

Even though we could stick to the original interface, it would be pointless:
No-one is using it and makes the implementation less clean. 
We'd need to transformations to obtain the necessary information, which would result
more overridden (and therefore diverging) methods in blotter. Not to mention that Law of Demeter would also be broken...